src/components/Map.vue

```diff
@@ -276,15 +275,39 @@ export default {
         });
       },
     },
+    isMrXVisible: {
+      handler() {
+        const svg = document.getElementById("board");
+        if (this.isMrXVisible) {
+          // check if user has access to Mr X
+          const isMrXTurn =
+            this.currentPlayer && this.currentPlayer.role === "mr-x";
+
+          if (!isMrXTurn && this.isMrXVisible && this.mrXLocation) {
+            const mrXLocationStation = this.$refs[this.mrXLocation][0];
+            const attributes = {
+              cx: mrXLocationStation.getAttributeNS(null, "cx"),
+              cy: mrXLocationStation.getAttributeNS(null, "cy"),
+              className: `mr-x-location`,
+            };
+            this.addCircleToBoard(attributes);
+          }
+        } else {
+          document.querySelectorAll("circle.mr-x-location").forEach(node => {
+            svg.removeChild(node);
+          });
+        }
+      },
+    },
     state: {
       deep: true,
       handler() {
-        this.checkIfMrXVisible();
+        this.setCurrentPlayer();
       },
     },
     currentPlayer: {
       handler() {
-        this.checkIfMrXVisible();
+        this.setCurrentPlayer();
       },
     },
   },
```

Comment:
Bro tbh not understanding, what these lines are doing? What is the handler function?

Reply:
`watch` is basically a Vue component helper that gets called whenever the watched property is changed. the handler is basically, the function that triggers when that happens. 

see: https://vuejs.org/v2/api/#watch

I can simplify this a lot by just removing the handler function since it's shorthand calls the function defined as well as seen in the doc above. 

but in plain english - basically when the `state` changes (user is logged in or not) as well as when the currentPlayer changes (at the end of each player's turn) we want to call the `setCurrentPlayer` function.